tutorials/ngsi_v2/e4_iot_thermal_zone_sensors/e4_iot_thermal_zone_sensors_solution.py

```diff
@@ -197,66 +201,68 @@
-    #  or the attribute name as key in your payload.
+    # ToDo: Create a loop that publishes a message every 100 milliseconds
+    #  to the broker that holds the simulation time `sim_time` and the
+    #  corresponding temperature `temperature`. You may use the `object_id`
+    #  or the attribute name as a key in your payload.
     for t_sim in range(sim_model.t_start,
                        sim_model.t_end + int(COM_STEP),
                        int(COM_STEP)):
         # publish the simulated ambient temperature
         mqttc.publish(device_id=weather_station.device_id,
                       payload={"temperature": sim_model.t_amb,
-                               "simtime": sim_model.t_sim})
+                               "sim_time": sim_model.t_sim})
 
-        # ToDo: publish the simulated zone temperature
+        # ToDo: Publish the simulated zone temperature.
         mqttc.publish(device_id=zone_temperature_sensor.device_id,
                       payload={"temperature": sim_model.t_zone,
-                               "simtime": sim_model.t_sim})
+                               "sim_time": sim_model.t_sim})
 
-        # simulation step for next loop
+        # simulation step for the next loop
         sim_model.do_step(int(t_sim + COM_STEP))
         # wait for one second before publishing the next values
-        time.sleep(1)
+        time.sleep(0.1)
 
-        # Get corresponding entities and write values to history
+        # get corresponding entities and store the data
         weather_station_entity = cbc.get_entity(
             entity_id=weather_station.entity_name,
             entity_type=weather_station.entity_type
         )
         # append the data to the local history
         history_weather_station.append(
-            {"simtime": weather_station_entity.simtime.value,
+            {"sim_time": weather_station_entity.sim_time.value,
              "temperature": weather_station_entity.temperature.value})
 
-        # ToDo: Get ZoneTemperatureSensor and write values to history
+        # ToDo: Get zone temperature sensor and store the data.
         zone_temperature_sensor_entity = cbc.get_entity(
             entity_id=zone_temperature_sensor.entity_name,
             entity_type=zone_temperature_sensor.entity_type
         )
         history_zone_temperature_sensor.append(
-            {"simtime": zone_temperature_sensor_entity.simtime.value,
+            {"sim_time": zone_temperature_sensor_entity.sim_time.value,
              "temperature": zone_temperature_sensor_entity.temperature.value})
 
     # close the mqtt listening thread
     mqttc.loop_stop()
     # disconnect the mqtt device
     mqttc.disconnect()
 
-    # plot results
+    # plot the results
     fig, ax = plt.subplots()
-    t_simulation = [item["simtime"] for item in history_weather_station]
+    t_simulation = [item["sim_time"]/3600 for item in history_weather_station]
```

Comment:
Here and in the plot below I changed the x-axis to hours for better readability.